libsql-sqlite3/src/pager.c

```diff
@@ -4920,28 +4911,7 @@ int sqlite3PagerOpen(
   }
 
 #ifndef SQLITE_OMIT_WAL
-  pPager->pWalMethods = pWalMethods;
-  /* Fill in Pager.zWal */
-  if( nWalPathname>0 ){
-    pPager->zWal = (char*)pPtr;
-    pWalMethods->xGetWalPathname((char *)pPtr, zPathname, nPathname);
-    pPtr += nWalPathname;
-    pPtr[0] = '\0';
-    pPtr++;
-#ifdef SQLITE_ENABLE_8_3_NAMES
-    sqlite3FileSuffix3(zFilename, pPager->zWal);
-    pPtr = (u8*)(pPager->zWal + sqlite3Strlen30(pPager->zWal)+1);
-#endif
-  }else{
-    pPager->zWal = 0;
-  }
-
-  if (pWalMethods->xPreMainDbOpen) {
-    int rc = pWalMethods->xPreMainDbOpen(pWalMethods, zPathname);
-    if (rc != SQLITE_OK) {
-      return rc;
-    }
-  }
+  memset(&(pPager->wal) ,0, sizeof(libsql_wal));
```

Comment:
that was the red flag that made me realized `pWal` became `wal` in this series. I think it should be rolled back to `pWal` to keep `Pager`'s size smaller and independent on `libsql_wal` size.